src/pages/index.js

```diff
@@ -74,25 +107,123 @@ const addImageModal = new ModalWithForms(
   config
 );
 
+// CREATE A MODALWITHFORM FOR CHANGE PICTURE
+const avatarEditModal = new ModalWithForms(
+  "#modal-change-picture",
+  handleAvatarFormSubmit,
+  config
+);
+
+
+// CREATE A MODALWITHFORM FOR DELETE CONFIRMATION
+const confirmationModal = new ModalWithConfirmation(
+  "#delete-confirmation-modal",
+{
+  submitButtonSelector: ".modal__button",
+  formSelector: ".modal__form",
+}
+);
+
+
+
 // FUNCTION PROFILE EDIT SUBMIT
-function handleProfileFormSubmit(values) {
-  userInfo.setUserInfo(values);
-  profileEditModal.close();
+function handleProfileFormSubmit(inputValues) {
+  profileEditModal.renderingSaving(true);
+  api
+    .editUserInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+      profileEditModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profileEditModal.renderingSaving(false);
+    });
 }
 
+
 // FUNCTION ADD IMAGE SUBMIT
+
 function handleAddImageFormSubmit(inputValues) {
-  const card = {
+  addImageModal.renderingSaving(true);
+  const newData = {
     name: inputValues.title,
-
     link: inputValues.url,
   };
-  const cardElement = createCard(card);
-  const addCardFormValidator = formValidators["card-form"];
-  cardsContainer.addItem(cardElement);
-  addCardFormValidator.disableSubmitButton();
-  addCardFormValidator.resetForm();
-  addImageModal.close();
+  api
+    .addCard(newData)
+    .then((res) => {
+      const cardElement = createCard(res);
+      formValidators.resetForm();
+      formValidators['add-card-form'].resetValidation();
+      cardsContainer.addItem(cardElement);
+      addImageModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      addImageModal.renderingSaving(false)
+    });
+}
+
+
+ 
+
+
+
+// DELETE CARD FUNCTION
+function handleDeleteClick(card) {
+  confirmationModal.renderingSaving(true);
```

Comment:
Needs correction: Please remove this line, you should start the loading before the api request on line 181